Automattic/jetpack
# Settings UI: hide settings for unavailable modules

Hide settings for all modules that are disabled through the `jetpack_get_available_modules` filter.

Fixes #5212

#### Changes proposed in this Pull Request:

- update reducer to initialize once, resolve ambiguity for searchTerm, optimize string matching, fail asap and return
- hide settings for disabled modules. Consolidate calls to avoid running isModuleFound twice since it's expensive.

#### Testing instructions:

* add code to disable modules and check that it's reflected in settings
```
add_filter( 'jetpack_get_available_modules', function( $modules ) {
	unset( $modules['masterbar'] );
	unset( $modules['after-the-deadline'] );
	unset( $modules['carousel'] );
	return $modules;
} );
```
